From 7b1a65925b9462b0a84fe6e840ff0c90b3b19bfc Mon Sep 17 00:00:00 2001
From: Doug Goldstein <cardoe@cardoe.com>
Date: Mon, 15 Dec 2025 16:35:57 -0600
Subject: [PATCH 3/8] fix redfish inspect system product name

The intention of this code is to read the system product name which is
stored in the model field per the Redfish spec and not in the name field
which will always store the name of the object we are working with. This
results in the value always being 'System'.

Closes-Bug: 2136233
Change-Id: I375fbe27253d7965e458be7b147d5b72cffa4e89
Signed-off-by: Doug Goldstein <cardoe@cardoe.com>
---
 ironic/drivers/modules/redfish/inspect.py                   | 4 ++--
 .../redfish-inspect-product-name-bd537f417d35b254.yaml      | 6 ++++++
 2 files changed, 8 insertions(+), 2 deletions(-)
 create mode 100644 releasenotes/notes/redfish-inspect-product-name-bd537f417d35b254.yaml

diff --git a/ironic/drivers/modules/redfish/inspect.py b/ironic/drivers/modules/redfish/inspect.py
index 59c3ce2b0..69f0f0c09 100644
--- a/ironic/drivers/modules/redfish/inspect.py
+++ b/ironic/drivers/modules/redfish/inspect.py
@@ -149,8 +149,8 @@ class RedfishInspect(base.InspectInterface):
                 inventory['interfaces'].append(iface)
 
         system_vendor = {}
-        if system.name:
-            system_vendor['product_name'] = str(system.name)
+        if system.model:
+            system_vendor['product_name'] = str(system.model)
 
         if system.serial_number:
             system_vendor['serial_number'] = str(system.serial_number)
diff --git a/releasenotes/notes/redfish-inspect-product-name-bd537f417d35b254.yaml b/releasenotes/notes/redfish-inspect-product-name-bd537f417d35b254.yaml
new file mode 100644
index 000000000..4b0b7c4b6
--- /dev/null
+++ b/releasenotes/notes/redfish-inspect-product-name-bd537f417d35b254.yaml
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    When using the redfish inspection driver, the system product name in the
+    data was always the value 'System' due to the wrong field being read.
+    See `bug 2136233 <https://bugs.launchpad.net/ironic/+bug/2136233>`_
-- 
2.50.1 (Apple Git-155)
